gson/src/test/java/com/google/gson/functional/PrimitiveTest.java

```diff
@@ -1031,20 +1036,6 @@ public void testValueVeryCloseToZeroIsZero() {
     assertThat(gson.fromJson("122.08e-2132", double.class)).isEqualTo(0.0);
   }
 
-  @Test
-  public void testDeserializingBigDecimalAsFloat() {
-    String json = "-122.08e-2132332";
```

Comment:
This is unfortunate. Do we need the logic in `JsonReader` or would it be enough to just have the new logic in `TypeAdapters` at the point where the number is converted to a `BigDecimal` or `BigInteger`?

Reply:
I mainly removed these tests because a few lines above something similar is covered already:
https://github.com/google/gson/blob/c98efd8288f708f5d040564033dc6bcbd7924cdd/gson/src/test/java/com/google/gson/functional/PrimitiveTest.java#L1033-L1036

To me it also does not look like this `...e-2132332` is any specific number but just a number with a large amount of exponent digits.
For comparison, `Double.MIN_VALUE` is 4.9e-324.

I assume it would be possible to adjust the logic in `JsonReader` and defer the check until `nextString()` is called, but that might make it a bit more complicated, possibly having to check the string there, or adding a new field indicating whether the parsed number exceeded the limits.
And because at least these numbers here are so contrived I am not sure if that is worth it.

Reply:
Oh, I misunderstood. I thought these tests were removed because these values were now rejected. If they still produce the same results but are redundant then it's fine to remove them.

Reply:
Sorry, you did understand it correctly; they are rejected now.

But I removed them instead of adjusting them (e.g. to use `...e-9999`) because in my opinion they did not add much value compared to the other tests: `122.08e-2132` is already parsed as `0.0`, so testing `-122.08e-2132332` seemed a bit pointless to me.

Do you think this should be supported though? I am not sure if these are really realistic values, or just dummy values for testing.

Reply:
I have removed the limit checks in `JsonReader` now because they would also be ineffective when `nextString()` is called but the JSON data contains a JSON string (for which no restrictions exist) instead of a JSON number, and the user did not explicitly use `peek()` to verify that the value is a JSON number.

However, the parsing logic for `Double.parseDouble` (which is called by `JsonReader.nextDouble`) and `Float.parseFloat` is quite complex, see https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/jdk/internal/math/FloatingDecimal.java
But if it had performance problems for certain number strings, then it would affect all other parts of Gson which parse as `double` or `float` as well.

Reply:
OK, so if I understand correctly, these tests would now pass, but they are redundant given `testValueVeryCloseToZeroIsZero` just above. The name of the method is `testDeserializingBigDecimalAsFloat` but actually it has nothing to do with `BigDecimal`. Is that right?

Reply:
Yes, that is right.